src/distPlus/distribution/DistTypes.re

```diff
@@ -14,20 +14,36 @@ type xyShape = {
   ys: array(float),
 };
 
+type interpolation = [
+    | `Stepwise
+    | `Linear
+];
+type extrapolation = [
+    | `UseZero
+    | `UseOutermostPoints
+];
+
+type interpolator = (xyShape, int, float) => float;
+
 type continuousShape = {
   xyShape,
-  interpolation: [ | `Stepwise | `Linear],
-  knownIntegralSum: option(float),
+  interpolation: interpolation,
+  integralSumCache: option(float),
+  integralCache: option(continuousShape),
 };
 
 type discreteShape = {
   xyShape,
-  knownIntegralSum: option(float),
+  /* interpolation is always `Discrete */
```

Comment:
Note that comments in types breaks autoformatting

Reply:
Ah nvm, this is from something I had tried earlier and isn't really needed anymore. I'm deleting it